Source/WebCore/editing/MarkupAccumulator.cpp

```diff
@@ -201,34 +203,72 @@ String MarkupAccumulator::serializeNodes(Node& targetNode, SerializedNodes root,
 
 void MarkupAccumulator::serializeNodesWithNamespaces(Node& targetNode, SerializedNodes root, const Namespaces* namespaces, Vector<QualifiedName>* tagNamesToSkip)
 {
-    if (tagNamesToSkip && is<Element>(targetNode)) {
-        for (auto& name : *tagNamesToSkip) {
-            if (downcast<Element>(targetNode).hasTagName(name))
-                return;
-        }
-    }
-
-    Namespaces namespaceHash;
+    WTF::Vector<std::unique_ptr<Namespaces>> namespaceStack;
```

Comment:
I think your rationale here is incorrect. An empty HashMap is just a pointer. See wtf/HashTable.h:

        union {
            ValueType* m_table { nullptr };
            unsigned* m_tableForLLDB;
        };

Therefore, unique_ptr<HashMap> is either a pointer (in the null case) or a pointer to a pointer (in the non-null case). That's pure overhead.

(You could be forgiven for not noticing this, since it is so heavily templated and abstracted.)

I recommend switching back to Vector<Namespaces>.

Reply:
Thanks I totally missed that. I switched it back